crates/valence_new/src/view.rs

```diff
@@ -0,0 +1,161 @@
+use glam::DVec3;
+use valence_protocol::BlockPos;
+
+/// The X and Z position of a chunk in an
+/// [`Instance`](crate::instance::Instance).
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Default, Hash, Debug)]
+pub struct ChunkPos {
+    /// The X position of the chunk.
+    pub x: i32,
+    /// The Z position of the chunk.
+    pub z: i32,
+}
+
+const EXTRA_VIEW_RADIUS: i32 = 2;
```

Comment:
What's this for?

Reply:
Chunks will only render if all the adjacent chunks are also loaded, so this value corrects for that. After some experimentation, 2 appears to be the correct value for this but I'm not certain. I'll add a comment explaining this.